File: VB/DxBlazorApplication1/Services/Interfaces/IResourceService.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DxBlazorApplication1.Services.Interfaces

    Public Interface IResourceService(Of TResource)

        Public Function GetResourcesAsync() As Task(Of IList(Of TResource))

    End Interface
End Namespace

File: CS/DxBlazorApplication1/Services/Interfaces/IResourceService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DxBlazorApplication1.Services.Interfaces {
    public interface IResourceService<TResource> {
        public Task<IList<TResource>> GetResourcesAsync();
    }
}
